src/components/grid-smart/grid-smart.tsx

```diff
@@ -0,0 +1,547 @@
+import {
+  Component,
+  ComponentInterface,
+  Element,
+  Event,
+  EventEmitter,
+  Listen,
+  Method,
+  Prop,
+  Watch
+} from "@stencil/core";
+
+import { GridBaseHelper, IGridBase } from "../grid-base/grid-base";
+import { IVisibilityComponent } from "../common/interfaces";
+// tslint:disable:no-duplicate-imports
+import Swiper from "swiper";
+import { SwiperOptions } from "swiper";
+
+@Component({
+  styleUrl: "grid-smart.scss",
+  tag: "gx-grid-smart"
+})
+export class GridSmart
+  implements IGridBase, ComponentInterface, IVisibilityComponent {
+  @Element() el!: HTMLElement;
+
+  private scrollbarEl?: HTMLElement;
+  private paginationEl?: HTMLElement;
+  private didInit = false;
+  private readySwiper!: (swiper: Swiper) => void;
+
+  private swiper: Promise<Swiper> = new Promise(resolve => {
+    this.readySwiper = resolve;
+  });
+
+  /**
+   * Number of items per view (items visible at the same time on slider's container).
+   */
+  @Prop() columns: number | "auto";
+
+  /**
+   * This attribute lets you specify how this element will behave when hidden.
+   *
+   * | Value        | Details                                                                     |
+   * | ------------ | --------------------------------------------------------------------------- |
+   * | `keep-space` | The element remains in the document flow, and it does occupy space.         |
+   * | `collapse`   | The element is removed form the document flow, and it doesn't occupy space. |
+   */
+  @Prop() invisibleMode: "collapse" | "keep-space" = "collapse";
+
+  /**
+   * Grid loading State. It's purpose is to know rather the Grid Loading animation or the Grid Empty placeholder should be shown.
+   *
+   * | Value        | Details                                                                                        |
+   * | ------------ | ---------------------------------------------------------------------------------------------- |
+   * | `loading` | The grid is waiting the server for the grid data. Grid loading mask will be shown.                |
+   * | `loaded`   | The grid data has been loaded. If the grid has no records, the empty place holder will be shown. |
+   */
+
+  @Prop() loadingState: "loading" | "loaded";
+
+  /**
+   * Set numbers of items to define and enable group sliding. Useful to use with rowsPerPage > 1
+   */
+  @Prop() itemsPerGroup: number;
+
+  /**
+   * Could be 'horizontal' or 'vertical' (for vertical slider).
+   */
+  @Prop() direction: "horizontal" | "vertical";
+  /**
+   * Advanced options to pass to the swiper instance.
+   * See http://idangero.us/swiper/api/ for valid options
+   */
+  @Prop() options: SwiperOptions = {};
```

Comment:
This is for advanced use only, just in case. 
Yes, it can be used as an HTML attribute:

<grid options="{prop1:1}">

Not using it right now from the Angular generator.